### ISSUE TITLE
[BUG] Tokenizer attention_mask is all-ones for a single string input

### ISSUE THREAD

**Issue by YeonwooSung** [pytorch/rl#4223]
## Describe the bug

`Tokenizer.call_tokenizer_fn` builds a real `attention_mask` from the Hugging Face tokenizer when the input is a `list[str]`, but for a single `str` it encodes with `tokenizer.encode` and then fills `torch.ones_like(out)`. With `max_length` padding, pad tokens are therefore marked as attended.

## To Reproduce

```170:197:torchrl/envs/llm/transforms/tokenizer.py
    def call_tokenizer_fn(self, value: str | list[str]):
        device = self.device
        kwargs = {"add_special_tokens": self.add_special_tokens}
        if self.max_length is not None:
            kwargs["padding"] = "max_length"
            kwargs["max_length"] = self.max_length
        if isinstance(value, str):
            out = self.tokenizer.encode(value, return_tensors="pt", **kwargs)[0]
            # TODO: incorporate attention mask
            if self.return_attention_mask:
                attention_mask = torch.ones_like(out, dtype=torch.int64)
        else:
            kwargs["padding"] = (
                self.padding if self.max_length is None else "max_length"
            )
            kwargs["return_attention_mask"] = self.return_attention_mask
            out = self.tokenizer(value, return_tensors="pt", **kwargs)
            if self.return_attention_mask:
                attention_mask = out["attention_mask"]
            out = out["input_ids"]
        ...
        if self.return_attention_mask:
            return out, attention_mask
```

The `# TODO: incorporate attention mask` is at line 178. The single-string branch never asks the tokenizer for a mask.

Sketch (any short string + `max_length` larger than the encoded length):

```python
from tensordict import TensorDict, NonTensorData
from torchrl.envs.llm.transforms.tokenizer import Tokenizer

tok = Tokenizer(
    in_keys=["text"],
    out_keys=["tokens"],
    max_length=16,
    return_attention_mask=True,
    add_special_tokens=False,
)

# single string -> all-ones mask, including pad positions
td_str = TensorDict({"text": NonTensorData("hi")}, batch_size=[])
out_str = tok(td_str)

# batched list[str] -> tokenizer's real mask (0 on pads)
td_list = TensorDict({"text": ["hi"]}, batch_size=[1])
out_list = tok(td_list)

print(out_str["attention_mask"])   # ones of length 16
print(out_list["attention_mask"])  # 1s on "hi" (+ specials), 0s on pads
```

`Tokenizer` is constructed with `return_attention_mask=True` by default (`tokenizer.py:64`).

## Expected behavior

Pad positions should be `0` on the single-string path, matching the `list[str]` path and the tokenizer's own `attention_mask`.

## Screenshots

N/A

## System info

Observed on a source checkout of pytorch/rl at `1d3de3db`. Repro needs `transformers` (the default tokenizer is `google-bert/bert-base-uncased` when none is passed).

## Additional context

Downstream losses that AND this mask with an assistant mask (`SFTLoss`, `DistillationLoss`, GRPO) will treat pad tokens as valid if the observation came in as a lone string.

## Reason and Possible fixes

Use the same `tokenizer(..., return_attention_mask=True)` path for `str` as for `list[str]` (e.g. wrap the string as `[value]` and squeeze), or call `tokenizer.encode` / `tokenizer` in a way that actually returns the mask instead of `torch.ones_like`. The existing TODO at line 178 is this fix.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)
